src/main/java/com/example/demo/entities/Subcategory.java

```diff
@@ -0,0 +1,56 @@
+package com.example.demo.entities;
+
+import java.io.Serializable;
+import javax.persistence.*;
+
+
+/**
+ * The persistent class for the "subcategories" database table.
+ * 
+ */
+@Entity
+@Table(name="subcategories")
+@NamedQuery(name="Subcategory.findAll", query="SELECT s FROM Subcategory s")
+public class Subcategory implements Serializable {
+	private static final long serialVersionUID = 1L;
+
+	@Id
+	@GeneratedValue(strategy=GenerationType.IDENTITY)
+	@Column(name="id")
+	private int id;
+
+	@ManyToOne
+	@JoinColumn(name="category")
+	private Category category;
+
+	@Column(name="name")
+	private String name;
```

Comment:
if only difference between category and subcategory is parent category, merge them into one database entity to not duplicate code